python/ray/tune/sync_client.py

```diff
@@ -583,39 +508,45 @@ def sync_up(
         self._last_source_tuple = source_ip, source
         self._last_target_tuple = target_ip, target_path
 
-        # Get existing files on remote node before packing on local node
-        self._last_files_stats = _remote_get_recursive_files_and_stats.options(
-            num_cpus=0, resources={f"node:{target_ip}": 0.01}
-        ).remote(target_path)
-
-        return self._execute_sync(
-            self._last_source_tuple, self._last_target_tuple, self._last_files_stats
-        )
+        return self._execute_sync(self._last_source_tuple, self._last_target_tuple)
 
     def _execute_sync(
         self,
         source_tuple: Tuple[str, str],
         target_tuple: Tuple[str, str],
-        files_stats: Optional[Dict[str, Tuple[float, int]]] = None,
     ) -> bool:
         source_ip, source_path = source_tuple
         target_ip, target_path = target_tuple
 
-        pack_on_source_node = _pack_dir.options(
-            num_cpus=0, resources={f"node:{source_ip}": 0.01}
-        )
-        unpack_on_target_node = _unpack_dir.options(
-            num_cpus=0, resources={f"node:{target_ip}": 0.01}
+        self._sync_future, pack_actor, files_stats = sync_dir_between_nodes(
+            source_ip=source_ip,
+            source_path=source_path,
+            target_ip=target_ip,
+            target_path=target_path,
+            _return_all_remotes=True,
         )
 
-        pack_future = pack_on_source_node.remote(source_path, files_stats)
-        if self._store_pack_future:
-            self._pack_future = pack_future
-        self._sync_future = unpack_on_target_node.remote(pack_future, target_path)
+        if self._store_remotes:
+            self._stored_pack_actor_ref = pack_actor
+            self._stored_files_stats = files_stats
+
         return True
 
     def delete(self, target: str):
-        pass
+        if not self._last_target_tuple:
+            logger.error(
+                f"Could not delete path {target} as we target node is not known."
+            )
+            return
+
+        node_ip = self._last_target_tuple[0]
+
+        try:
+            ray.get(delete_on_node(node_ip=node_ip, path=target))
+        except Exception as e:
+            logger.error(
+                f"Could not delete path {target} on remote node {node_ip}: {e}"
+            )
 
     def wait(self):
```

Comment:
Since we can sync arbitrarily large file size now, we may run into this method more often than before and end up in a blocking situation. We need to think of how to supply visibility if/when this happens.

Reply:
Fwiw, we synced arbitrarily large files before as well, just with rsync and not with remote tasks. And we do warn already: `with warn_if_slow("callbacks.on_trial_save")`, though we may want to think about making this message a bit more insightful